src/main/java/org/opentripplanner/client/model/LegGeometry.java

```diff
@@ -0,0 +1,3 @@
+package org.opentripplanner.client.model;
+
+public record LegGeometry(String points) {}
```

Comment:
This is generally ok, but in the end I want to use a [JTS `LineString`](https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/LineString.html).

Either you add the parsing in this PR or you will have to live with the breakage whenever I touch this part next.

Reply:
Actually, it will probably enough if you encapsulate this well enough:

- Add https://gitlab.com/leonard.ehrenfried/google-polyline-codec to this project
- Convert this to a class
- Add getter methods for the Google-encoded string and the list of points (converted on the fly)

That way client applications can choose what they want. If I need a JTS conversion I can add it myself.

Reply:
Sounds good. We also use the JTS linestring.
I just wasn't sure because some people probably don't need the lineString and I wasn't sure if it might affect the performance.

But your advice is the best of both worlds.

Reply:
I changed it by adding a toLinestring method.